pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -202,31 +202,51 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
-// getSeccompProfileFromAnnotations gets seccomp profile from annotations.
-// It gets pod's profile if containerName is empty.
-func (m *kubeGenericRuntimeManager) getSeccompProfileFromAnnotations(annotations map[string]string, containerName string) string {
-	// try the pod profile.
-	profile, profileOK := annotations[v1.SeccompPodAnnotationKey]
+func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]string, containerName string,
+	podSecContext *v1.PodSecurityContext, containerSecContext *v1.SecurityContext) string {
+	fieldProfile := func(scmp *v1.SeccompProfile) string {
+		if scmp.Type == v1.SeccompProfileTypeRuntimeDefault {
+			return v1.SeccompProfileRuntimeDefault
+		}
+		if scmp.Type == v1.SeccompProfileTypeLocalhost && scmp.LocalhostProfile != nil {
+			fname := filepath.Join(m.seccompProfileRoot, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname
+		}
+		return v1.SeccompProfileNameUnconfined
```

Comment:
I'd prefer an exact match here rather than defaulting an unknown type to "unconfined":
```go
if scmp.Type == v1.SeccompProfileTypeUnconfined {
  return v1.SeccompProfileNameUnconfined
}
return ""
```

Reply:
Now it is all aligned with existing behaviour - non-expected state returns "".